libs/filesystem/src/main/java/org/padaiyal/utilities/filesystem/FileSystemUtility.java

```diff
@@ -0,0 +1,692 @@
+
+/**
+ * File utility library for creating, modifying, deleting and monitoring files and directories.
+ */
+public class FileSystemUtility {
+
+  /**
+   * Logger object to log information and errors.
+   */
+  private static final Logger logger = LogManager.getLogger(FileSystemUtility.class);
+
+  /**
+   * Map of paths to the WatchRegistrationInfo, used to store events to track for each path.
+   */
+  private static final ConcurrentHashMap<String, WatchServiceRegistrationInfo>
+      pathToWatchServiceRegistrationInfoMap = new ConcurrentHashMap<>();
+
+  /**
+   * Map that stores the path that triggered an event and the event type they generated.
+   */
+  private static final ConcurrentHashMap<String, Queue<Kind<?>>> pathToTriggeredEventMap =
+      new ConcurrentHashMap<>();
+
+  /**
+   * Thread object that visits all files to look for events.
+   */
+  private static Thread backgroundWatchServiceRegisteringThread;
+
+  /**
+   * Runnable used by the backgroundWatchServiceRegisteringThread.
+   */
+  public static final Runnable backgroundWatchServiceRegisteringThreadRunnable = () -> {
+
+    while (Boolean.parseBoolean(
+            PropertyUtility.getProperty(
+                "FileSystemUtility.backgroundThread.watchService.switch"))) {
+      HashMap<String, WatchServiceRegistrationInfo>
+          pathToWatchServiceRegistrationInfoMapCopy =
+          new HashMap<>(pathToWatchServiceRegistrationInfoMap);
+
+      // Iterate through all registered watch services and check if any events are
+      // generated
+      pathToWatchServiceRegistrationInfoMapCopy.keySet()
+          .stream()
+          .sorted()
+          .forEach(
+              (subPath) -> {
+
+                WatchServiceRegistrationInfo subPathWatchServiceInfo =
+                        pathToWatchServiceRegistrationInfoMapCopy
+                                .get(subPath);
+
+                if (subPathWatchServiceInfo != null) {
+                  WatchService subPathWatchService = subPathWatchServiceInfo
+                          .getWatchService();
+
+                  try {
+                    WatchKey watchKey = subPathWatchService.poll();
+
+                    if (watchKey != null) {
+                      List<WatchEvent<?>> watchEvents = watchKey.pollEvents();
+                      /* Iterate through all watch events and add another watch service
+                      for folders created within the maxDepth level. */
+                      for (WatchEvent<?> event : watchEvents) {
+
+                        Path changePath =
+                                Paths.get(subPath).resolve((Path) event.context());
+
+                        String changePathString = changePath.toAbsolutePath().toString();
+
+                        if (StandardWatchEventKinds.ENTRY_CREATE.equals(event.kind())) {
+
+                          if (Files.isDirectory(changePath.toAbsolutePath())) {
+
+                            @SuppressWarnings("UnusedAssignment")
+
```

Comment:
Try running the auto linter in your IDE with the google check style config.
https://github.com/google/google-java-format#intellij-android-studio-and-other-jetbrains-ides

Reply:
Didn't notice this, was in the original PR.